### PR TITLE
Add robotraconteur package source to jazzy/distribution.yaml and rolling/distribution.yaml

### PR DESCRIPTION
**Pull request by johnwason**
Please add the following dependency to the rosdep database.

## Package name:

robotraconteur

## Package Upstream Source:

https://github.com/robotraconteur/robotraconteur/tree/ros

## Purpose of using this:

Robot Raconteur is a communication framework with an ecosystem containing a number of standardized industrial robot drivers, and a universal teach pendant under development.

https://github.com/robotraconteur/robotraconteur-directory
https://github.com/pyri-project/pyri-core/blob/master/README.md

Having the core `robotraconteur` package with C++ and Python bindings in rosdep will make integration with ROS easier.

This package was previously accepted to ROS Noetic in #31844

# Checks
 - [x] All packages have a declared license in the package.xml
 - [x] This repository has a LICENSE file
 - [x] This package is expected to build on the submitted rosdistro

## Notes

This PR modifies the other distributions to use the consolidated `ros` branch in the `robotraconteur/robotraconteur` repository.

### CODE REVIEW

File: humble/distribution.yaml [hunk 1]
```diff
@@ -7508,7 +7508,7 @@ repositories:
     source:
       type: git
       url: https://github.com/robotraconteur/robotraconteur.git
-      version: ros2-humble
+      version: ros
```
Comment (sloretz) [inline humble/distribution.yaml:7511]: Is this change intentional? If so you'll want to also change the branch in the tracks in the release repository

* https://github.com/robotraconteur-packaging/robotraconteur-ros2-release/blob/5a683d18afb5aa54bdb838a413f929ce3e4e9ada/tracks.yaml#L17
* https://github.com/robotraconteur-packaging/robotraconteur-ros2-release/blob/5a683d18afb5aa54bdb838a413f929ce3e4e9ada/tracks.yaml#L44

File: noetic/distribution.yaml [hunk 2]
```diff
@@ -9346,7 +9346,7 @@ repositories:
     source:
       type: git
       url: https://github.com/robotraconteur/robotraconteur.git
-      version: ros-noetic
+      version: ros
```
Comment (sloretz) [inline noetic/distribution.yaml:9349]: Same question about this change:

https://github.com/robotraconteur-packaging/robotraconteur-ros-release/blob/ce2b79da28408d28083121be912bfcce246589a1/tracks.yaml#L17

File: humble/distribution.yaml (hunk 1 above)
Comment (johnwason) [reply humble/distribution.yaml:7511]: I am going to do another release in the next few days that will update the release repository.

File: noetic/distribution.yaml (hunk 2 above)
Comment (johnwason) [reply noetic/distribution.yaml:9349]: See above

**Review by sloretz — APPROVED** [review]
